hedera-mirror-rosetta/app/services/account_service_test.go

```diff
@@ -0,0 +1,183 @@
+		blockIdentifier = &rTypes.PartialBlockIdentifier{
+			Index: &index,
+			Hash:  &hash,
+		}
+	}
+	return &rTypes.AccountBalanceRequest{
+		AccountIdentifier: &rTypes.AccountIdentifier{Address: "0.0.1"},
+		BlockIdentifier:   blockIdentifier,
+	}
+}
+
+func TestNewAccountAPIService(t *testing.T) {
+	mocks.Setup()
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	assert.IsType(t, &AccountAPIService{}, accountService)
+}
+
+func TestAccountBalance(t *testing.T) {
+	// given:
+	exampleBlock := &types.Block{
+		Index:               1,
+		Hash:                "0x123jsjs",
+		ConsensusStartNanos: 1000000,
+		ConsensusEndNanos:   20000000,
+		ParentIndex:         2,
+		ParentHash:          "0xparenthash",
+	}
+
+	expectedAccountBalanceResponse := &rTypes.AccountBalanceResponse{
+		BlockIdentifier: &rTypes.BlockIdentifier{
+			Index: 1,
+			Hash:  "0x123jsjs",
+		},
+		Balances: []*rTypes.Amount{
+			{
+				Value:    "1000",
+				Currency: config.CurrencyHbar,
+			},
+		},
+	}
+
+	exampleAmount := &types.Amount{
+		Value: int64(1000),
+	}
+
+	mocks.Setup()
+	mocks.MBlockRepository.On("RetrieveLatest").Return(exampleBlock, mocks.NilError)
+	mocks.MAccountRepository.On("RetrieveBalanceAtBlock").Return(exampleAmount, mocks.NilError)
+
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	// when:
+	actualResult, e := accountService.AccountBalance(nil, exampleRequest(false))
+
+	// then:
+	assert.Equal(t, expectedAccountBalanceResponse, actualResult)
+	assert.Nil(t, e)
+}
+
+func TestAccountBalanceFullData(t *testing.T) {
+	// given:
+	exampleBlock := &types.Block{
+		Index:               1,
+		Hash:                "0x123jsjs",
+		ConsensusStartNanos: 1000000,
+		ConsensusEndNanos:   20000000,
+		ParentIndex:         2,
+		ParentHash:          "0xparenthash",
+	}
+
+	expectedAccountBalanceResponse := &rTypes.AccountBalanceResponse{
+		BlockIdentifier: &rTypes.BlockIdentifier{
+			Index: 1,
+			Hash:  "0x123jsjs",
+		},
+		Balances: []*rTypes.Amount{
+			{
+				Value:    "1000",
+				Currency: config.CurrencyHbar,
+			},
+		},
+	}
+
+	exampleAmount := &types.Amount{
+		Value: int64(1000),
+	}
+
+	mocks.Setup()
+	mocks.MBlockRepository.On("FindByIdentifier").Return(exampleBlock, mocks.NilError)
+	mocks.MAccountRepository.On("RetrieveBalanceAtBlock").Return(exampleAmount, mocks.NilError)
+
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	// when:
+	actualResult, e := accountService.AccountBalance(nil, exampleRequest(true))
+
+	//then:
+	assert.Equal(t, expectedAccountBalanceResponse, actualResult)
+	assert.Nil(t, e)
+	mocks.MBlockRepository.AssertNotCalled(t, "RetrieveLatest")
+}
+
+func TestAccountBalanceThrows(t *testing.T) {
```

Comment:
- We could add another test that checks that when `RetrieveBlock` returns an error, we are returning that error, similar to this test but for `FindByIdentifier`
- I think that `TestAccountBalanceThrowsWhenRetrieveLatestFails` is a better name for the test + we would have `TestAccountBalanceThrowsWhenRetrieveBlockFails` test as-well :)